libcloud/container/utils/docker.py

```diff
@@ -0,0 +1,179 @@
+
+        :param  max_count: The maximum number of records to return
+        :type   max_count: ``int``
+
+        :return: A list of images
+        :rtype: ``list`` of :class:`libcloud.container.base.ContainerImage`
+        """
+        path = '/v2/repositories/%s/%s/tags/?page=1&page_size=%s' \
+               % (namespace, repository_name, max_count)
+        response = self.connection.request(path)
+        images = []
+        for image in response.object['results']:
+            images.append(self._to_image(repository_name, image))
+        return images
+
+    def get_repository(self, repository_name, namespace='library'):
+        """
+        Get the information about a specific repository
+
+        :param  repository_name: The name of the repository e.g. 'ubuntu'
+        :type   repository_name: ``str``
+
+        :param  namespace: (optional) The docker namespace
+        :type   namespace: ``str``
+
+        :return: The details of the repository
+        :rtype: ``object``
+        """
+        path = '/v2/repositories/%s/%s' % (namespace, repository_name)
+        response = self.connection.request(path)
+        return response.object
+
+    def get_image(self, repository_name, tag='latest', namespace='library'):
+        """
+        Get an image from a repository with a specific tag
+
+        :param repository_name: The name of the repository, e.g. ubuntu
+        :type  repository_name: ``str``
+
+        :param  tag: (optional) The image tag (defaults to latest)
+        :type   tag: ``str``
+
+        :param  namespace: (optional) The docker namespace
+        :type   namespace: ``str``
+
+        :return: A container image
+        :rtype: :class:`libcloud.container.base.ContainerImage`
+        """
+        path = '/v2/repositories/%s/%s/tags/%s' \
+               % (namespace, repository_name, tag)
+        response = self.connection.request(path)
+        return self._to_image(repository_name, response.object)
+
+    def _to_image(self, repository_name, obj):
+        path = '%s/%s:%s' % (self.connection.host,
+                             repository_name,
+                             obj['name'])
+        return ContainerImage(
+            id=obj['id'],
+            path=path,
+            name=path,
+            version=obj['name'],
+            extra={
+                'full_size': obj['full_size']
+            },
+            driver=None
+        )
+
+
+class HubClient(RegistryClient):
+    """
+    A client for the Docker Hub API
+
+    The hub is based on the v2 registry API
+    """
+    host = 'registry.hub.docker.com'
+
+    def __init__(self, username=None, password=None, **kwargs):
+        """
+        Construct a Docker hub client
+
+        :param username: (optional) Your Hub account username
+        :type  username: ``str``
+
+        :param password: (optional) Your hub account password
+        :type  password: ``str``
+        """
+        super(HubClient, self).__init__(self.host, username,
```

Comment:
Minor thing - please use keyword arguments when calling methods when possible.

This means less surprises in case the method signature changes or similar.